Userland/Services/AudioServer/FadingProperty.h

```diff
@@ -17,11 +17,11 @@ constexpr int DEFAULT_FADE_TIME = 10;
 template<typename T>
 class FadingProperty {
 public:
-    FadingProperty(T const value)
+    FadingProperty(T value)
```

Comment:
idk what `T` can be, but if we're moving it into m_new_value below, seems we should `move` it here as well into the constructor invocation. Or if it's always numeric, this should have a constraint & the `move` into `m_new_value` below should be removed.

Reply:
From what I can tell they are only `doubles`
@kleinesfilmroellchen can you elaborate what the intended and actual use are?

Reply:
FadingProperty will move to LibDSP eventually since it's a generally useful concept for mixing and parameter control. There, it will also be used to smoothly transition with floats, fixed-point numbers and samples (it's actually quite ideal for compressor-like filters). So I hope it's okay to keep it general for now, until I move it to LibDSP once we need it there.

Reply:
So everything is trivially copyable, if I understand that correctly, so I can safely remove all complaining moves and `const&`s without worry

Reply:
> idk what `T` can be, but if we're moving it into m_new_value below, seems we should `move` it here as well into the constructor invocation.

Yes.

> Or if it's always numeric, this should have a constraint & the `move` into `m_new_value` below should be removed.

Also yes! The constraint, as mentioned above, should be a specific requires clause since I'd like Samples to be included. That basically means the type must be arithmetic (support + and *) and be trivially copyable/movable (and default constructable). I'm pretty sure we don't have a concept for all of those, but I'm sure we can come up with something or use a `requires requires` clause.